queue/queue.go

```diff
@@ -0,0 +1,75 @@
+package queue
+
+import (
+	"sync"
+	"time"
+)
+
+// Build is a request waiting for a worker
+type Build struct {
+	Pipeline string `json:"pipeline"`
+	Manifest string `json:"manifest"`
+}
+
+// Job is an image build already in progress
+type Job struct {
+	UUID     string `json:"uuid"`
+	WorkerID string `json:"worker-id"`
+	Build    Build  `json:"build"`
+}
+
+// JobQueue contains already running jobs waiting for
+type JobQueue struct {
+	sync.Mutex
+	incomingBuilds chan Build
+	queue          []Job
+	timeoutSeconds int
```

Comment:
I suggest dropping any timeout handling for now, and we can figure out how to deal with it in a future PR. I imagine reusing the PATCH call with "status": "running" at regular intervals from the worker may do the trick.